cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlannerTest.java

```diff
@@ -122,60 +126,110 @@ public void testGetInterBrokerPartitionMovementTasks() {
     ExecutionTaskPlanner prioritizeSmallMovementPlanner = new ExecutionTaskPlanner(null, new KafkaCruiseControlConfig(prioritizeSmallMovementProps));
 
+    // Create smallUrpMovementPlanner
+    Properties smallUrpMovementProps = KafkaCruiseControlUnitTestUtils.getKafkaCruiseControlProperties();
+    smallUrpMovementProps.setProperty(ExecutorConfig.DEFAULT_REPLICA_MOVEMENT_STRATEGIES_CONFIG,
+                                      String.format("%s,%s", PrioritizeSmallReplicaMovementStrategy.class.getName(),
+                                                    PostponeUrpReplicaMovementStrategy.class.getName()));
+    ExecutionTaskPlanner smallUrpMovementPlanner = new ExecutionTaskPlanner(null, new KafkaCruiseControlConfig(smallUrpMovementProps));
+
+    // Create contradictingMovementPlanner containing both small and large replica movements
+    Properties contradictingMovementProps = KafkaCruiseControlUnitTestUtils.getKafkaCruiseControlProperties();
+    contradictingMovementProps.setProperty(ExecutorConfig.DEFAULT_REPLICA_MOVEMENT_STRATEGIES_CONFIG,
+                                           String.format("%s,%s,%s", PrioritizeSmallReplicaMovementStrategy.class.getName(),
+                                                         PostponeUrpReplicaMovementStrategy.class.getName(),
+                                                         PrioritizeLargeReplicaMovementStrategy.class.getName()));
+    ExecutionTaskPlanner contradictingMovementPlanner = new ExecutionTaskPlanner(null, new KafkaCruiseControlConfig(contradictingMovementProps));
+
+    // Create prioritizeMinIsrMovementPlanner
+    Properties prioritizeMinIsrMovementProps = KafkaCruiseControlUnitTestUtils.getKafkaCruiseControlProperties();
+    prioritizeMinIsrMovementProps.setProperty(ExecutorConfig.DEFAULT_REPLICA_MOVEMENT_STRATEGIES_CONFIG,
+                                              PrioritizeMinIsrWithOfflineReplicasStrategy.class.getName());
+    ExecutionTaskPlanner prioritizeMinIsrMovementPlanner = new ExecutionTaskPlanner(null,
+                                                                                    new KafkaCruiseControlConfig(prioritizeMinIsrMovementProps));
+
+
     Set<PartitionInfo> partitions = new HashSet<>();
-    partitions.add(generatePartitionInfo(_partitionMovement1, true));
+    partitions.add(generatePartitionInfoWithUrpHavingOfflineReplica(_partitionMovement1, true));
     partitions.add(generatePartitionInfo(_partitionMovement2, false));
-    partitions.add(generatePartitionInfo(_partitionMovement3, true));
+    partitions.add(generatePartitionInfoWithUrpHavingOfflineReplica(_partitionMovement3, true));
     partitions.add(generatePartitionInfo(_partitionMovement4, false));
 
     Cluster expectedCluster = new Cluster(null, _expectedNodes, partitions, Collections.<String>emptySet(), Collections.<String>emptySet());
+    // This ensures that the _partitionMovement1 and _partitionMovement3 are AtMinISR, while the other partitions are not.
```

Comment:
Maybe we want to add a test for UnderMinISR is prioritized before AtMinISR partitions?
I am a little hesitant because I don't see a quick way to add this without involving 3 replicas in the replica set.